py/desispec/scripts/zproc.py

```diff
@@ -608,7 +609,12 @@ def main(args=None, comm=None):
         qafile = findfile('tileqa', **findfileopts)
         qapng = findfile('tileqapng', **findfileopts)
         qalog = findfile('tileqa', logfile=True, **findfileopts)
+        ## requires all coadd and redrock outputs in addition to exposureqa
         infiles = []
+        for expid, night in zip(expids, nights):
+            findfileopts['expid'] = expid
+            findfileopts['night'] = night
+            infiles.append(findfile('exposureqa', **findfileopts))
```

Comment:
This loop has the side effect of leaving `findfileopts['night']` to be the last entry in the `nights` list (similar for expid).  That happens to be the same as the original `findfileopts['night'] = thrunight` value if `groupname='cumulative'`, but that depends upon how `nights` and `thrunight` were constructed hundreds of lines earlier.  That could change in the future without realizing the impact here.

This is admittedly not the only place that modifies `findfileopts` on the fly (and those other places likely trace back to me), but it still feels especially fragile here since it is touching nights that aren't the thrunight of the zproc.  Since `findfile('exposureqa', ...)` only needs `night` and `expid` and doesn't need the other variables that are currently set in `findfileopts`, I suggest bypassing that with something like
```python
for expid, night in zip(expids, nights):
    infiles.append(findfile('exposureqa', night=night, expid=expid, readonly=True))
```
or otherwise making a copy of `findfileopts` into a different dictionary that is modified on the fly without changing the original `findfileopts` and risking leaving it with a different night and expid than before.